analyses/molecular-subtyping-integrate/00-prepare-extraction-table.Rmd

```diff
@@ -0,0 +1,87 @@
+---
+title: "00-prepare-extraction-table"
+author: "Jo Lynne Rokita"
+date: 2022
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## This notebook wrangles manually curated files which have been collated to designate whether a tuomor DNA and RNA specimen came from the same extraction, same tissue chunk ("same tissue aliquot"), or different tissue chunk ("different tissue aliquot"). 
+
+
+## Load library/input files
+```{r}
+library(tidyverse)
+
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+input_dir <- file.path(root_dir,
+                      "analyses",
+                      "molecular-subtyping-integrate",
+                       "input")
+
+results_dir <- file.path(root_dir,
+                         "analyses",
+                         "molecular-subtyping-integrate",
+                         "results")
+
+# Read in PNOC and CBTN metadata files
+pnoc_aliquots <- read_tsv(file.path(input_dir, "pnoc_aliquots.tsv"))
+cbtn_aliquots <- read_tsv(file.path(input_dir, "cbtn_aliquots.tsv")) %>%
+  # necessary for combining with PNOC later
+  mutate(aliquot_id = as.character(aliquot_id))
+```
+
+# Get extraction status from PNOC cohort
+```{r}
+# Using the parental_aliquot_id, determine which DNA + RNA pair together (ie came from the same extraction).
+pnoc_aliquots_status <- pnoc_aliquots %>%
+  # split sample and aliquot id by experimental strategy to group by parental aliquot id later
+  mutate(sample_id_rna = ifelse(experimental_strategy == "RNA-Seq", sample_id, ""),
+         sample_id_wxs = ifelse(experimental_strategy == "WXS", sample_id, ""),
+         sample_id_wgs = ifelse(experimental_strategy == "WGS", sample_id, ""),
+         aliquot_id_rna = ifelse(experimental_strategy == "RNA-Seq", aliquot_id, ""),
+         aliquot_id_wxs = ifelse(experimental_strategy == "WXS", aliquot_id, ""),
+         aliquot_id_wgs = ifelse(experimental_strategy == "WGS", aliquot_id, "")) %>%
+  distinct() %>%
+  select(-c(experimental_strategy, sample_id, aliquot_id)) %>%
+  group_by(parent_aliquot_id) %>%
+  # collapse
+  summarise_all(funs(trimws(paste(., collapse = '')))) %>%
+  # if there is an RNA sample and DNA sample which have the same parental aliquot id, deem them as from the same extraction
+  mutate(extraction_type = ifelse(aliquot_id_rna != "" & (aliquot_id_wxs != "" | aliquot_id_wgs != ""),
+                                    "Same extraction", NA_character_)) %>%
+  mutate_all(na_if,"") %>%
+  # rename to be compatible with CBTN data
+  dplyr::rename(parental_aliquot_id = parent_aliquot_id)
+```
+
+# Combine CBTN and PNOC RNA sample and aliquot ids with extraction status
+```{r}
+combined <- pnoc_aliquots_status %>%
+  # only keep those with RNA-Seq (had extraction annotated)
+  filter(!is.na(extraction_type)) %>%
```

Comment:
I think this is going to miss one case of RNA-seq (sample id `7316-4998`) that has `NA` for `extraction_type`. I assume that means it won't be from the same extraction?

Reply:
Oh, yes at this point, there is one RNA-Seq which had no matched DNA, so it doesn't have an extraction type

Reply:
Should we include that though? Either as an `NA` or something else descriptive like `No matched DNA sample`

Reply:
It becomes NA in the final histology file, but yeah, maybe we should include here, since there are other NAs. There are also 20 which it seems Noel couldn't find and so they were not propagated into my list - Let me add those here as NA also so we can have all 1035 RNA ids in that extraction file.

Reply:
@jaclyn-taroni I update this to not only include this one, but add the remaining 20. Turns out, there are two samples with the same sample and aliquot ids, so the total is now 1033 in the final file.